File: examples/SymInferASE/nla/Ps3.java
public class Ps3 {

     public static void main (String[] args) {
	  mainQ(Integer.parseInt(args[0]));
     }

     public static int mainQ(int k){
	  assert (k>=0);
	  assert (k<=30);
     
	  int y = 0;
	  int x = 0;
	  int c = 0;


	  while(true){
	       //assert(6*x-2*y*y*y-3*y*y-y == 0);	  
	       //%%%traces: int x, int y, int k

	       if (!(c < k)) break;    
	       c = c +1 ;
	       y=y +1;
	       x=y*y+x;
	  }
	  return x;
     }
}

File: old/dig1/dig/dig2/programs/nla/ps3.c
#include <stdio.h>
#include <assert.h>
#include <stdlib.h>  //required for afloat to work


int mainQ(int k){
  int y = 0;
  int x = 0;
  int c = 0;

  while(1){
    if (!(c < k)) break;
    //%%%traces: int x, int y, int k
    c = c +1 ;
    y=y +1;
    x=y*y+x;
  }
  return x;
}



int main(int argc, char **argv){
  mainQ(atoi(argv[1]));
  return 0;
}

